codebase2/codebase-sqlite/sql/001-temp-entity-tables.sql

```diff
@@ -0,0 +1,61 @@
+create table temp_entity_type_description (
+  id integer primary key not null,
+  description text unique not null
+);
+insert into temp_entity_type_description values
+  (0, 'Term Component'),
+  (1, 'Decl Component'),
+  (2, 'Namespace'),
+  (3, 'Patch'),
+  (4, 'Causal');
+
+-- A "temp entity" is a term/decl/namespace/patch/causal that we cannot store in the database proper due to missing
+-- dependencies.
+--
+-- The existence of each `temp_entity` row implies the existence of one or more corresponding
+-- `temp_entity_missing_dependency` rows: it does not make sense to make a `temp_entity` row for a thing that has no
+-- missing dependencies!
+--
+-- Similarly, each `temp_entity` row implies we do not have the entity in the database proper. When and if we *do* store
+-- an entity proper (after storing all of its dependencies), we should always atomically delete the corresponding
+-- `temp_entity` row, if any.
+create table temp_entity (
+  hash text primary key not null,
```

Comment:
Could you explain the reasoning to not use hash_id's here? This table should generally be small, so it's not a space constraint or anything, and it's probably easier to work with this way anyways, I'm just curious if it was thought about and what the decision was 😄